warehouse/query-core/src/test/java/datawave/query/jexl/visitors/FieldMissingFromSchemaVisitorTest.java

```diff
@@ -0,0 +1,328 @@
+        actual = FieldMissingFromSchemaVisitor.getNonExistentFields(helper, script, null, specialFields);
+        assertEquals(Collections.emptySet(), actual);
+        
+        // Case 3 - check with an empty datatype filter, implies all fields.
+        actual = FieldMissingFromSchemaVisitor.getNonExistentFields(helper, script, Collections.emptySet(), specialFields);
+        assertEquals(Collections.emptySet(), actual);
+    }
+    
+    /**
+     * Test query with function for a field that does not exist in the schema.
+     */
+    @Test
+    public void testRegexFunctionWithUnknownField() throws ParseException {
+        String query = "filter:includeRegex(FOO, 'bar.*')";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        
+        // Setup MockMetadataHelper with emptyFieldsToDatatypes
+        MockMetadataHelper helper = new MockMetadataHelper();
+        helper.addFieldsToDatatypes(HashMultimap.create());
+        
+        // Setup datatype filter
+        Set<String> dataTypeFilter = Sets.newHashSet("datatype1");
+        
+        // Case 1 - check with a datatype filter
+        Set<String> actual = FieldMissingFromSchemaVisitor.getNonExistentFields(helper, script, dataTypeFilter, specialFields);
+        Set<String> expected = Sets.newHashSet("FOO");
+        assertEquals(expected, actual);
+        
+        // Case 2 - check with a null datatype filter, implies all fields.
+        actual = FieldMissingFromSchemaVisitor.getNonExistentFields(helper, script, null, specialFields);
+        assertEquals(expected, actual);
+        
+        // Case 3 - check with an empty datatype filter, implies all fields.
+        actual = FieldMissingFromSchemaVisitor.getNonExistentFields(helper, script, Collections.emptySet(), specialFields);
+        assertEquals(expected, actual);
+    }
+    
+    /**
+     * Test query with function for a field that does not exist in the schema under the provided datatype filter.
+     */
+    @Test
+    public void testRegexFunctionWithUnknownFieldBecauseOfDatatypeFilter() throws ParseException {
+        String query = "filter:includeRegex(FOO, 'bar.*')";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        
+        // Fields to datatypes for our MetadataHelper
+        Multimap<String,String> fieldsToDatatypes = HashMultimap.create();
+        fieldsToDatatypes.put("FOO", "datatype1");
+        
+        // Setup MockMetadataHelper
+        MockMetadataHelper helper = new MockMetadataHelper();
+        helper.addFieldsToDatatypes(fieldsToDatatypes);
+        
+        // Setup datatype filter
+        Set<String> dataTypeFilter = Sets.newHashSet("datatypeX");
+        
+        // Case 1 - check with a datatype filter
+        Set<String> actual = FieldMissingFromSchemaVisitor.getNonExistentFields(helper, script, dataTypeFilter, specialFields);
+        Set<String> expected = Sets.newHashSet("FOO");
+        assertEquals(expected, actual);
```

Comment:
Shouldn't the dataTypeFilter being set to 'datatypeX' have prevented FOO of type 'datatype1' from being returned?  It looks like the EvaluationPhaseFilterFunctionsDescriptor ignored the dataTypeFilter that's passed in, so fields of all types can be returned.  Is this desired, or did you uncover a bug?

Reply:
This looks like a bug.